File: backend/src/main/java/com/cdac/farmermarketplace/service/OrderService.java
package com.cdac.farmermarketplace.service;

import java.util.List;

import com.cdac.farmermarketplace.dto.request.PlaceOrderRequest;
import com.cdac.farmermarketplace.dto.response.OrderResponse;


public interface OrderService {


    // Place order for logged-in user
    OrderResponse placeOrder(
            PlaceOrderRequest request,
            Long userId
    );



    // Get order by id
    OrderResponse getOrderById(
            Long orderId,
            Long userId
    );



    // Get logged-in user's orders
    List<OrderResponse> getOrdersByUserId(
            Long userId
    );



    // Cancel order
    void cancelOrder(
            Long orderId,
            Long userId
    );


}

File: frontend/src/services/orderService.js
import api from "./api";

const orderService = {

  // Place Order
  placeOrder: async (orderData) => {
    const response = await api.post("/orders/place", orderData);
    return response.data;
  },

  // Get Single Order
  getOrder: async (orderId) => {
    const response = await api.get(`/orders/${orderId}`);
    return response.data;
  },

  // Get Logged-in User Orders
  getMyOrders: async () => {
    const response = await api.get("/orders/my-orders");
    return response.data;
  },

  // Cancel Order
  cancelOrder: async (orderId) => {
    const response = await api.put(`/orders/cancel/${orderId}`);
    return response.data;
  }

};

export default orderService;